src/smartbanner.js

```diff
@@ -106,11 +107,43 @@ export default class SmartBanner {
   }
 
   get buttonUrl() {
-    if (this.platform === 'android') {
-      return this.options.buttonUrlGoogle;
-    } else if (this.platform === 'ios') {
-      return this.options.buttonUrlApple;
+    let options = this.options;
+    let device = this.device;
+
+    if (this.ignoreIOSGlobalLink === 'iosglobal') {
+      options.buttonUrlApple = '';
+    }
+
+    if (this.ignoreAndroidGlobalLink === 'androidglobal') {
+      options.buttonUrlGoogle = '';
+    }
+
+    if (this.ignoreButtonUrlAppleIpad === 'ipad') {
+      options.buttonUrlAppleIpad = '';
+    }
+
+    if (this.ignoreButtonUrlAppleIphone === 'iphone' || this.ignoreButtonUrlAppleIpod === 'ipod') {
+      options.buttonUrlAppleIphone = '';
     }
+
+    if (this.ignoreButtonUrlGooglePhone === 'phone') {
+      options.buttonUrlGooglePhone = '';
+    }
+
+    if (this.ignoreButtonUrlGoogleTablet === 'tablet') {
+      options.buttonUrlGoogleTablet = '';
+    }
+
+    if (device === 'ipad') {
+      return options.buttonUrlAppleIpad ? options.buttonUrlAppleIpad : options.buttonUrlApple;
+    } else if (device === 'iphone' || device === 'ipod') {
+      return options.buttonUrlAppleIphone ? options.buttonUrlAppleIphone : options.buttonUrlApple;
+    } else if (device === 'phone') {
+      return options.buttonUrlGooglePhone ? options.buttonUrlGooglePhone : options.buttonUrlGoogle;
+    } else if (device === 'tablet') {
+      return options.buttonUrlGoogleTablet ? options.buttonUrlGoogleTablet : options.buttonUrlGoogle;
+    }
+
```

Comment:
I tested this today in the office on our iPad mini 4 and it worked perfectly fine for me. Also my implementation was used in the project, which I am working on and our QA team also verified that all 4 cases are working correctly, so I am quite surprised that it is not working for you.
Can you do a check of your userAgent and return the result? [http://www.whoishostingthis.com/tools/user-agent/](url)